### PR TITLE
chore(Storybook): Search with Algolia support

### PR DESCRIPTION
**Pull request by frassinier**
**What is the problem this PR is trying to solve?**
We can't search for specific content into Docs

**What is the chosen solution to this problem?**
Add Algolia support

**Please check if the PR fulfills these requirements**

- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Changelog has been commited, e.g: `yarn changeset`
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -108,6 +109,10 @@ export const parameters = {
 
 			return (
 				<>
+					<Helmet>
+						<title>{props.context.title?.split('/').join(' / ')}</title>
```
Comment (frassinier) [inline .storybook/preview.js:113]:
```suggestion
						<title>{props.context.title?.replace('/', ' / ')}</title>
```

File: .storybook/preview-head.html [hunk 2]
```diff
@@ -210,3 +210,15 @@
...
 </style>
+
+<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@algolia/algoliasearch-netlify-frontend@1/dist/algoliasearchNetlify.css" />
+<script type="text/javascript" src="https://cdn.jsdelivr.net/npm/@algolia/algoliasearch-netlify-frontend@1/dist/algoliasearchNetlify.js"></script>
+<script type="text/javascript">
+	algoliasearchNetlify({
+		appId: '9MAEU8VF6E',
+		apiKey: '1ccc24ce1685b14139d476bc26f79802',
+		siteId: 'd6d66424-7754-4257-bb5e-cc6de2f9d9aa',
+		branch: 'frassinier/chore/storybook/algolia' || 'master',
```
Comment (frassinier) [inline .storybook/preview-head.html:221]:
```suggestion
		branch: 'master',
```

File: .storybook/main.js [hunk 3]
```diff
@@ -48,6 +48,7 @@ module.exports = {
 	addons: [
 		'@storybook/addon-a11y',
 		'@storybook/addon-essentials',
+		'@storybook/addon-interactions',
```
Comment (frassinier) [inline .storybook/main.js:51]:
```suggestion
```

File: .storybook/preview.js [hunk 4]
```diff
@@ -106,16 +107,23 @@ export const parameters = {
...
 			}, [hasBootstrapStylesheet]);
 
+			const title = props.context.title;
+			const titleArray = title?.split('/');
+
 			return (
 				<>
+					<Helmet>
+						<title>{title?.replace(/\//, ' / ')}</title>
+						{titleArray.length > 1 && <meta property="article:section" value={titleArray[1]} />}
```
Comment (frassinier) [inline .storybook/preview.js:117]:
```suggestion
						{titleArray.length > 1 && <meta property="article:section" value={titleArray[0]} />}
```

File: netlify.toml [hunk 5]
```diff
@@ -2,6 +2,16 @@
   ignore = "git log -1 --pretty=%B | grep dependabot"
 
 [[plugins]]
-  package = "@algolia/netlify-plugin-crawler"
+package = "@algolia/netlify-plugin-crawler"
+
+    [plugins.inputs]
+    branches = ['master', 'frassinier/chore/storybook/algolia']
```
Comment (frassinier) [inline netlify.toml:8]:
```suggestion
    branches = ['master']
```

File: src/docs/Algolia.component.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+import { Item } from './AlgoliaItem.component';
+
+const searchClient = algoliasearch('9MAEU8VF6E', '1ccc24ce1685b14139d476bc26f79802');
+
+function Algolia() {
+	return (
+		<div style={{ maxWidth: '33%', margin: '0 auto' }}>
+			<Autocomplete
+				openOnFocus
+				placeholder="Search for..."
```
Comment (frassinier) [inline src/docs/Algolia.component.js:15]:
```suggestion
				placeholder="Search for design system item..."
```

File: src/docs/Algolia.component.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+				getSources={({ query }) => [
+					{
+						sourceId: 'pages',
+						getItems() {
+							return getAlgoliaResults({
+								searchClient,
+								queries: [
+									{
+										indexName:
+											'netlify_d6d66424-7754-4257-bb5e-cc6de2f9d9aa_frassinier-chore-storybook-algolia_all',
```
Comment (frassinier) [inline src/docs/Algolia.component.js:25]:
```suggestion
											'netlify_d6d66424-7754-4257-bb5e-cc6de2f9d9aa_master_all',
```

File: .storybook/preview-head.html [hunk 8]
```diff
@@ -209,4 +209,26 @@
...
+
+	.aa-ItemLink,
+	.aa-ItemLink:hover,
+	.aa-ItemLink:focus,
+	.aa-ItemLink:active {
+		text-decoration: none;
+	}
+
+	.aa-ItemContentTitle,
+	.aa-ItemContentCategory {
```
Comment (frassinier) [inline .storybook/preview-head.html:221]:
```suggestion
	.aa-ItemContentTitle {
```

File: .storybook/preview.js [hunk 9]
```diff
@@ -106,16 +107,36 @@ export const parameters = {
 					.forEach(link => (link.disabled = !hasBootstrapStylesheet));
 			}, [hasBootstrapStylesheet]);
 
+			const title = props.context.title;
+			const titleArray = title?.split('/');
+			const fTitle = title?.replaceAll(/\//gi, ' / ');
```
Comment (frassinier) [inline .storybook/preview.js:112]:
```suggestion
			const documentTitle = title?.replaceAll(/\//gi, ' / ');
```

File: .storybook/preview.js [hunk 10]
```diff
@@ -106,16 +107,36 @@ export const parameters = {
...
+
+			console.log(props.context);
+
 			return (
 				<>
+					<Helmet>
+						<title>{fTitle}</title>
```
Comment (frassinier) [inline .storybook/preview.js:119]:
```suggestion
						<title>{documentTitle}</title>
```

File: .storybook/preview.js [hunk 11]
```diff
@@ -106,16 +107,36 @@ export const parameters = {
...
+						<meta property="og:title" content={titleArray[titleArray.length - 1]} />
+						<meta property="og:type" content="article" />
+						<meta
+							property="og:url"
+							content={`https://design.talend.com/?path=/docs/${props.context.id}`}
+						/>
+						<meta
+							property="og:image"
+							content={`https://via.placeholder.com/450x100?text=${fTitle}`}
```
Comment (frassinier) [inline .storybook/preview.js:128]:
```suggestion
							content={`https://via.placeholder.com/450x100?text=${documentTitle}`}
```

File: .storybook/preview-head.html [hunk 12]
```diff
@@ -209,4 +209,25 @@
...
+		font-weight: bold;
+	}
+
+	.aa-ItemContentCategory {
+		color: #ccc;
+	}
+
+	.aa-ItemContentDescription {
+		overflow: hidden;
+	}
```
Comment (Benrajalu) [inline .storybook/preview-head.html:230]: You seem to be overriding stuff from Algolia here but your link import comes afterwards?

File: netlify.toml (hunk 5 above)
Comment (Benrajalu) [reply netlify.toml:7]: Yup :D

File: scripts/sitemap.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+			`  <url><loc>https://design.talend.com/iframe.html?id=${id}&amp;viewMode=docs</loc></url>`,
+	)
+	.join('\n')}
+</urlset>`;
+
+fs.writeFile('./storybook-static/sitemap.xml', sitemap, err => {
+	if (err) {
+		console.error(err);
+	}
+});
```
Comment (Benrajalu) [inline scripts/sitemap.js:37]: If we put all of this in your `try` we don't need a `let json;` right?

File: src/docs/SearchBar.component.js [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+		console.warn(
+			'Algolia is not set up correctly, be sure to configure STORYBOOK_ALGOLIA_SEARCH_APP_ID, STORYBOOK_ALGOLIA_SEARCH_API_KEY and STORYBOOK_ALGOLIA_SEARCH_INDEX secrets before.',
+		);
+		return null;
+	}
+
+	return (
+		<div style={{ maxWidth: '50%', margin: '0 auto' }}>
+			<Autocomplete
+				placeholder="Search for design system topic..."
```
Comment (Benrajalu) [inline src/docs/SearchBar.component.js:23]:
```suggestion
				placeholder="Search for anything on Coral"
```

File: scripts/sitemap.js (hunk 13 above)
Comment (frassinier) [reply scripts/sitemap.js:37]: Obviously, but all errors will end up in the catch 🤷🏻

File: .storybook/preview-head.html (hunk 12 above)
Comment (frassinier) [reply .storybook/preview-head.html]: Fixed

File: scripts/sitemap.js (hunk 13 above)
Comment (Benrajalu) [reply scripts/sitemap.js:37]: Which honestly seems fine by me, but I don't mind either way

**Review by Benrajalu — APPROVED** [review]

**Pull request merged**